### PR TITLE
APP-31433: Add `SidePanel` in Design System

### PR DESCRIPTION
**Pull request by habxtech**
See [APP-31433](https://habxfr.atlassian.net/browse/APP-31433): ui-core: Add SidePanel in DS.

### CODE REVIEW

File: src/SidePanel/SidePanel.interface.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import * as React from 'react'
+
+export interface SidePanelProps {
+  onClose: () => void
+  navigationContent?: React.ReactNode
+  tabsBarContent?: React.ReactNode
+  title: string
```
Comment (jean9696) [inline src/SidePanel/SidePanel.interface.ts]: Why `title` is required ?

File: src/SidePanel/SidePanel.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import { Text } from '../Text'
+
+import { SidePanel } from './index'
+
+const Wrapper = styled.div`
+  min-height: 500px;
+  max-height: 100vh;
+`
+
+const Container = styled.div`
```
Comment (jean9696) [inline src/SidePanel/SidePanel.stories.tsx]: I think we should use `Background` here

File: src/SidePanel/SidePanel.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  display: flex;
+  position: absolute;
+  top: 0;
+  bottom: 0;
+  left: 0;
+  right: 0;
+  opacity: 0;
+  &[data-open='true'] {
```
Comment (jean9696) [inline src/SidePanel/SidePanel.stories.tsx]: Why do you take the whole space for a **side** panel ? Plus, I think we should add an animation for the entrance

File: src/SidePanel/SidePanel.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    opacity: 1;
+  }
+`
+
+const Button = styled(BaseButton)`
+  z-index: 101;
```
Comment (jean9696) [inline src/SidePanel/SidePanel.stories.tsx]: Please use the `zIndex` constant

File: src/SidePanel/SidePanel.tsx [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+import { SidePanelProps } from './SidePanel.interface'
+import { CloseButton, Container, Header, Tabs } from './SidePanel.style'
+
+export const SidePanel: React.FunctionComponent<SidePanelProps> = ({
+  onClose,
+  tabsBarContent,
+  title,
+  navigationContent,
+}) => {
```
Comment (jean9696) [inline src/SidePanel/SidePanel.tsx]: I think that this component should have the same API than the `Modal` & `Lighbox` and use `useModal`

**Review by jean9696 — CHANGES_REQUESTED** [review]

File: src/SidePanel/SidePanel.interface.ts (hunk 1 above)
Comment (Carolinevp) [reply src/SidePanel/SidePanel.interface.ts]: There is no option without title in the [Figma](https://www.figma.com/file/PM3oYKBDgKXsdRds0tcJ5a/%F0%9F%A7%A9----Slide-Panel?node-id=1%3A106) so I figured it was mandatory

**Review by jean9696 — APPROVED** [review]

**Pull request merged**